commands/util/binary.js

```diff
@@ -0,0 +1,215 @@
+            .setDescription("Pick a binary number to convert into decimals")
+            .setRequired(true)
+        )
+    )
+    .addSubcommand(subcommand =>
+      subcommand
+        .setName("add")
+        .setDescription("Add two binary numbers")
+        .addIntegerOption(option =>
+          option
+            .setName("first")
+            .setDescription("The first number")
+            .setRequired(true)
+        )
+        .addIntegerOption(option =>
+          option
+            .setName("second")
+            .setDescription("The second number")
+            .setRequired(true)
+        )
+    )
+    .addSubcommand(subcommand =>
+      subcommand
+        .setName("subtract")
+        .setDescription("Subtract two binary numbers")
+        .addIntegerOption(option =>
+          option
+            .setName("first")
+            .setDescription("The first number")
+            .setRequired(true)
+        )
+        .addIntegerOption(option =>
+          option
+            .setName("second")
+            .setDescription("The second number")
+            .setRequired(true)
+        )
+    )
+    .addSubcommand(subcommand =>
+      subcommand
+        .setName("multiply")
+        .setDescription("Multiply two binary numbers")
+        .addIntegerOption(option =>
+          option
+            .setName("first")
+            .setDescription("The first number")
+            .setRequired(true)
+        )
+        .addIntegerOption(option =>
+          option
+            .setName("second")
+            .setDescription("The second number")
+            .setRequired(true)
+        )
+    )
+    .addSubcommand(subcommand =>
+      subcommand
+        .setName("divide")
+        .setDescription("Divide two binary numbers")
+        .addIntegerOption(option =>
+          option
+            .setName("first")
+            .setDescription("The first number")
+            .setRequired(true)
+        )
+        .addIntegerOption(option =>
+          option
+            .setName("second")
+            .setDescription("The second number")
+            .setRequired(true)
+        )
+    ),
+
+  async execute(interaction) {
+    try {
+      const subcommand = interaction.options.getSubcommand();
+
+      const validateBinary = (number) => /^[01]+$/.test(number.toString());
+
+      if (subcommand === "encode") {
+        const integer = interaction.options.getInteger("number");
+        const binary = integer.toString(2);
+
+        const embed = new EmbedBuilder()
+          .addFields(
+            { name: "Expression", value: `${integer}`, inline: true },
+            { name: "Result", value: `${binary}`, inline: true }
+          );
+
+        await interaction.reply({ embeds: [embed] });
+
+      } else if (subcommand === "decode") {
+        const binary = interaction.options.getInteger("binary").toString();
+        if (!validateBinary(binary)) {
+          throw new Error("Input must be a valid binary number.");
```

Comment:
Don't throw an error because that's going to crash the bot.

Send an ephemeral message that lets the user know that the input has to be a valid binary number.

Reply:
Also here you don't need to tell the user that an error occurred. Make sure the embed styling sends that message. Just have the text be "Input must be a valid binary number."